velox/common/memory/SharedArbitrator.cpp

```diff
@@ -258,6 +258,31 @@ SharedArbitrator::SharedArbitrator(const Config& config)
                       << participantConfig_.toString();
 }
 
+void SharedArbitrator::shutdown() {
+  {
+    std::lock_guard<std::mutex> l(stateLock_);
+    VELOX_CHECK(globalArbitrationWaiters_.empty());
+    if (shutdown_) {
+      return;
+    }
+    shutdown_ = true;
```

Comment:
Do we want to set the flag after shudownGlobalArbitration() call to be semantically consistent?

Reply:
This is to prevent we do shutdown twice